frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -300,18 +303,16 @@ function QueryBuilder(props) {
     return () => window.removeEventListener("resize", forceUpdateDebounced);
   });
 
-  const isExistingModelDirty = useMemo(
-    () => isModelQueryDirty || isMetadataDirty,
-    [isMetadataDirty, isModelQueryDirty],
-  );
+  const shouldShowUnsavedChangesWarningForModels =
+    isEditingModel && (isDirty || isMetadataDirty);
+  const shouldShowUnsavedChangesWarningForSqlQuery =
+    isNativeEditorOpen && isQuestionEdited;
```

Comment:
Why do we check that the native editor is opened (expanded) here? It doesn't change the fact my changes are lost. Should we check that the question is native (`question.isNative()`) instead?

Reply:
That's a good point. I used existing logic which also uses `isNativeEditorOpen`, but I can see if we open the editor edit the query then close the editor, there would be no browser prompt when we refresh the page.